tests/core/lcp/test_credential.py

```diff
@@ -34,27 +33,42 @@ def test_hashing(self):
         assert expected == received
 
 
-class TestCredentialFactory(DatabaseTest):
-    def setup_method(self):
-        super().setup_method()
-
-        self._factory = LCPCredentialFactory()
-        self._patron = self._patron()
-        self._data_source = DataSource.lookup(
-            self._db, DataSource.INTERNAL_PROCESSING, autocreate=True
+class ExampleCredentialFixture:
+    factory: LCPCredentialFactory
+    patron: Patron
+    data_source: DataSource
+    transaction: DatabaseTransactionFixture
+
+    @classmethod
+    def create(
+        cls, transaction: DatabaseTransactionFixture
+    ) -> "ExampleCredentialFixture":
+        data = ExampleCredentialFixture()
+        data.transaction = transaction
+        data.factory = LCPCredentialFactory()
+        data.patron = transaction.patron()
+        data.data_source = DataSource.lookup(
+            transaction.session(), DataSource.INTERNAL_PROCESSING, autocreate=True
         )
+        return data
+
+
+@pytest.fixture()
+def example_credential_fixture(
+    database_transaction: DatabaseTransactionFixture,
```

Comment:
Purely from a legacy nomenclature perspective should we leave the session names as `db` or `_db` as it is in the application code

Reply:
OK!

I hope the IDE can do this renaming. :crossed_fingers:

Reply:
Renamed the database transaction fixture to `db`, and added type annotations to test methods.